aspnetcore/fundamentals/host/generic-host.md

```diff
@@ -1,31 +1,494 @@
+* The [URLs](#urls) setting in web apps. The host passes this value to the servers through an `IServerAddresses` API.
+
+Host configuration automatically flows to app configuration ([ConfigureAppConfiguration](#configureappconfiguration).
+
+To add host configuration, call <xref:Microsoft.Extensions.Hosting.HostBuilder.ConfigureHostConfiguration*> on `IHostBuilder`. `ConfigureHostConfiguration` can be called multiple times with additive results. The host uses whichever option sets a value last on a given key.
+
+The environment variable provider with prefix `DOTNET_` is included by default. For web apps, the environment variable provider with prefix `ASPNETCORE_` is added. The prefix is removed when the environment variables are read. For example, the environment variable value for `ASPNETCORE_ENVIRONMENT` becomes the host configuration value for the `environment` key.
+
+The following example creates host configuration:
+
+[!code-csharp[](generic-host/samples-snapshot/3.x/Program.cs?name=snippet_HostConfig)]
+
+> [!TIP]
+> During development when using [Visual Studio](https://visualstudio.microsoft.com) or running an app with `dotnet run`, environment variables may be set in the *Properties/launchSettings.json* file. In [Visual Studio Code](https://code.visualstudio.com/), environment variables may be set in the *.vscode/launch.json* file. For more information, see <xref:fundamentals/environments>.
+
+## App configuration
+
+App configuration is created by calling <xref:Microsoft.Extensions.Hosting.HostBuilder.ConfigureAppConfiguration*> on `IHostBuilder`. `ConfigureAppConfiguration` can be called multiple times with additive results. The app uses whichever option sets a value last on a given key. 
+
+The configuration created by `ConfigureAppConfiguration` is available at [HostBuilderContext.Configuration](xref:Microsoft.Extensions.Hosting.HostBuilderContext.Configuration*) for subsequent operations and as a service from DI.
+
+App configuration automatically receives host configuration provided by `ConfigureHostConfiguration`.
+
+The following example creates app configuration:
+
+[!code-csharp[](generic-host/samples-snapshot/3.x/Program.cs?name=snippet_AppConfig)]
+
+> [!TIP]
+> To move settings files to the output directory for publishing, specify the settings files as [MSBuild project items](/visualstudio/msbuild/common-msbuild-project-items) in the project file. The sample app moves its JSON app settings files and *hostsettings.json* with the following `<Content>` item:
+> ```xml
+> <ItemGroup>
+>   <Content Include="**\*.json" Exclude="bin\**\*;obj\**\*" 
+>     CopyToOutputDirectory="PreserveNewest" />
+> </ItemGroup>
+> ```
+
+## Settings for all app types
+
+This section lists host settings that apply to non-HTTP as well as HTTP workloads. By default, environment variables used to configure these settings can have a `DOTNET_` or `ASPNETCORE_` prefix.
```

Comment:
This section lists host settings that apply to non-HTTP and HTTP workloads. By default, environment variables used to configure these settings can have a `DOTNET_` or `ASPNETCORE_` prefix.